### ISSUE TITLE
[Bug?]: Code navigation to a cell implementation not possible (ends up in type definition)

### ISSUE THREAD

**Issue by Philzen** [redwoodjs/redwood#5862]
### What's not working?

As a developer i'd like to be able to CTRL+click on any react component to pull up its implementation code.

This works fine in VSCode for everything but cells, likely caused by the fact that they don't have a matching default export.

#### Related

* Code completion not available for Cells, see #4303
* In other IDEs there is a problem when using the short path to a component, see #5097
* Details of the current resolution implementation at #234

### How do we reproduce the bug?

1. Generate a Cell, let's say `MyCell`
2. import and use it in any other component (e.g. a page)
3. Hold CTRL while clicking on `<MyCell … />` in your IDE of choice

The IDE opens `.redwood/types/mirror/web/src/components/MyCell/index.d.ts` instead of `web/src/components/MyCell/MyCell.tsx`.

### What's your environment? (If it applies)

```shell
System:
    OS: Linux 5.15 Manjaro Linux
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 16.15.1 - /tmp/xfs-c559b1ba/node
    Yarn: 3.2.0 - /tmp/xfs-c559b1ba/yarn
  Databases:
    SQLite: 3.38.5 - /usr/bin/sqlite3
  Browsers:
    Firefox: 102.0
  npmPackages:
    @redwoodjs/core: ^1.5.1 => 1.5.1
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by Philzen** [comment]
> I'm not aware of any solutions for this at this time, but will be very happy if there are any suggestions!

You already you were already on to something in your [tape](https://s.tape.sh/5LN35yUU) that you posted over at https://github.com/redwoodjs/redwood/issues/4303

> closing this as its a dupe of https://github.com/redwoodjs/redwood/issues/5097

I was worried about that at first before i opened this ticket, that's why i made some tests with WebStorm and these are clearly two separate issues :monocle_face: – although they are in the same domain (webpack, typemapping, default exports and stuff).

#5097 is about **Components not resolving correctly in IDEs other than VSCode** when using the "short" import path – yes that is *all* components, Cells are actually not mentioned in that issue.

That's why i opened this issue to specifically look at the challenge with cells, which are a different beast because there is no default export that would reference the cell itself. If at all, this could be considered a duplicate of #4303, and that ticket's scope should be widened – but the code navigation issue (this ticket) which may or *may not* be solved when #4303 is solved (depending on how it is solved). Therefore it may make sense to keep track of the different aspects separately.

**Comment by Tobbe** [comment]
The solution that I hinted at over in #4303 was specific to Cells IIRC. Unfortunately I prototyped the solution in GitPod, and it was lost in a browser crash a long time ago 🙁 

**If** my memory serves me right the idea was to generate a file somewhere inside .redwood/ that imported the Cell in a way that "educates" VSCode about its existence. 

At the time I had no idea about the complexities involved with keeping generated files in sync with actual files. But now I'm painfully aware of it all due to https://github.com/redwoodjs/redwood/pull/5820

After https://github.com/redwoodjs/redwood/pull/5820 is merged I can try to implement my now lost solution again, and see if it still holds up.
